ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -70,44 +71,62 @@ class Endpoints(
       lc: LoggingContextOf[InstanceUUID],
       metrics: Metrics,
   ): PartialFunction[HttpRequest, Http.IncomingConnection => Future[HttpResponse]] = {
-    val dispatch: PartialFunction[HttpRequest, LoggingContextOf[
-      InstanceUUID with RequestID
-    ] => Future[HttpResponse]] = {
-      // Parenthesis are required because otherwise scalafmt breaks.
+    type DispatchFun =
+      PartialFunction[HttpRequest, LoggingContextOf[InstanceUUID with RequestID] => Future[HttpResponse]]
+    // Parenthesis are required because otherwise scalafmt breaks.
+    val commandDispatch: DispatchFun = ({
       case req @ HttpRequest(POST, Uri.Path("/v1/create"), _, _, _) =>
         (implicit lc => httpResponse(create(req)))
       case req @ HttpRequest(POST, Uri.Path("/v1/exercise"), _, _, _) =>
         (implicit lc => httpResponse(exercise(req)))
       case req @ HttpRequest(POST, Uri.Path("/v1/create-and-exercise"), _, _, _) =>
         (implicit lc => httpResponse(createAndExercise(req)))
+    }: DispatchFun) andThen (f =>
+      lc => Timed.future(metrics.daml.HttpJsonApi.commandSubmissionTimer, f(lc))
+    )
+    val queryDispatch: DispatchFun = ({
       case req @ HttpRequest(POST, Uri.Path("/v1/fetch"), _, _, _) =>
         (implicit lc => httpResponse(fetch(req)))
       case req @ HttpRequest(GET, Uri.Path("/v1/query"), _, _, _) =>
         (implicit lc => httpResponse(retrieveAll(req)))
       case req @ HttpRequest(POST, Uri.Path("/v1/query"), _, _, _) =>
         (implicit lc => httpResponse(query(req)))
+    }: DispatchFun) andThen (f => lc => Timed.future(metrics.daml.HttpJsonApi.queryTimer, f(lc)))
```

Comment:
I don't remember off the top of my mind: what happens is a request is upgrade to `ws`? Is it handled here and would we start timing an arbitrary long WebSocket call, possibly polluting the stats?

Reply:
No need to worry here, web socket calls are handled seperate with different endpoints (see the `HttpServer.scala`).